test/gtfs/data_test.exs

```diff
@@ -0,0 +1,121 @@
+defmodule Gtfs.DataTest do
+  use ExUnit.Case, async: true
+
+  alias Gtfs.Data
+  alias Gtfs.Route
+  alias Gtfs.RoutePattern
+  alias Gtfs.StopTime
+  alias Gtfs.Trip
+
+  test "all_routes/1 returns all the routes" do
+    data = %Data{
+      routes: [%Route{id: "1"}, %Route{id: "2"}],
+      route_patterns: [],
+      stops: [],
+      trip_stop_times: %{},
+      trips: []
+    }
```

Comment:
(Applies to all the tests in this file)

Instead of creating the internal struct here, it might be nice to test something like
```ex
{:mocked_files, %{"routes.txt": [...]}
|> parse_files()
|> all_routes()
```
so the test doesn't depend on the internal structure, and is more similar to how it's used. I think it'd even mean you can put the struct in a private `defpstruct`

Reply:
I'm not sure I follow. This isn't internal sturcture, this is what Data is.

Reply:
Instead of testing that given some gtfs files, parsing creates the Data we expect; and then that given some Data, the query generates the answer we expect; we could do it it in one step by testing that given some gtfs files, parsing and then querying generates the answer we expect.

This would make the tests less dependent on the implementation details of the `Data.t()` format, but typing the gtfs files in the test might be more annoying than typing the data structs.

Reply:
It might be more annoying, and I also think that for everything other than the parse function we don't want to be thinking in terms of those files. Those are meaningless to us once they are parsed, we are operating on our own data struct for everything once we are going.